recipes/wavelet_buffer/all/test_package/conanfile.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from conan import ConanFile
+from conan.tools.build import cross_building
+from conan.tools.cmake import CMake
+from conan.tools.layout import cmake_layout
+
+
+class HelloTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    # VirtualBuildEnv and VirtualRunEnv can be avoided if "tools.env.virtualenv:auto_use" is defined
+    # (it will be defined in Conan 2.0)
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualBuildEnv", "VirtualRunEnv"
+    apply_env = False
+
```

Comment:
```suggestion

    def requirements(self):
        self.requires(self.tested_reference_str)

```
With Conan v2 migration, we need to adopt explicit requirement. 
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#changes-in-the-test-package-recipe